### ISSUE TITLE
v1.10.0 breaks command line `--ignore-pattern` behavior

### ISSUE THREAD

**Issue by sanemat** [eslint/eslint#4507]
v1.9.0 and earlier accepts command line option `--ignore-pattern` with `{}` like `--ignore-pattern "foo-{bar,baz}.js"`.
But v1.10.0 or newer does not accept `--ignore-pattern "foo-{bar,baz}.js"`.
Both v1.9.0 and v1.10.0 accepts `--ignore-pattern "foo-bar.js"`.

Is this expected behavior?

```
# with v1.10.0
$ npm run pattern:a

> @ pattern:a /Users/sane/work/tmp/eslint-ignore-pattern/not-work
> eslint --ignore-pattern "raise-{warning,foo}.js" .


/Users/sane/work/tmp/eslint-ignore-pattern/not-work/raise-warning.js
  1:5  error  "foo" is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

#### reproducing repository

https://github.com/sanemat/eslint-ignore-pattern

```
$ git ls-files
.gitignore
.gitkeep
latest/.eslintrc
latest/package.json
latest/raise-warning.js
not-work/.eslintrc
not-work/package.json
not-work/raise-warning.js
work/.eslintrc
work/package.json
work/raise-warning.js
```

```
$ cat not-work/package.json 
{
  "devDependencies": {
    "eslint": "1.10.0"
  },
  "scripts": {
    "pattern:a": "eslint --ignore-pattern \"raise-{warning,foo}.js\" .",
    "pattern:b": "eslint --ignore-pattern \"raise-warning.js\" ."
  }
}
```

```
my env:
osx 10.9.5
nodejs 4.2.1
```

**Comment by nzakas** [comment]
A change was made to allow me than one pattern to be passed, but that shouldn't have caused such a change. Definitely a bug.

**Comment by gyandeeps** [comment]
Update:

I think its a problem with `optionator` project

if this is the argument provide `"eslint --ignore-pattern \"raise-{warning,foo}.js\" ."`
Optionator creates an array of ignore patterns in this format
```
[
"raise-{warning",
"foo}.js"
]
```

Since we switched the option definition from `string` to `[string]` array of string

**Comment by nzakas** [comment]
@gkz is there any way to avoid splitting the option value on the comma?

**Comment by nzakas** [comment]
@sanemat while we are trying to get this straightened out, you can work around the issue by putting a slash before the comma ("\,").

**Comment by gyandeeps** [comment]
I dont think backslash before the comma `\,`would fix the problem.  What you can do right now is
```sh
eslint --ignore-pattern \"raise-warning.js\" --ignore-pattern \"raise-foo.js\" .
```

**Comment by gkz** [comment]
@nzakas I'll look into it

**Comment by gkz** [comment]
I can fix the problem, but it would mean that you wouldn't be able to create an array with `--ignore-pattern a,b`, but would need to use `--ignore-pattern` twice, eg. `--ignore-pattern a --ignore-pattern b` - since it can't know whether the comma is for delimiting values, or just part of your value. Is that acceptable?

**Comment by gyandeeps** [comment]
Thats fine, as that how we accept to form an array in this option as well as for other options.

**Comment by nzakas** [comment]
@gkz sounds good. Would it be possible to apply that just to this one option? Or would it need to be for all options?

**Comment by gkz** [comment]
I plan on making it a per-option option, so it would only affect `--ignore-pattern`.
This week is the last week of the term, so if you guys can wait one week for the fix, that would be great. If it is super urgent, let me know and I'll try to do something earlier.

**Comment by nzakas** [comment]
We can wait, thanks!
